Forward ported from qemu-2.11.1-capstone_include_path.patch.
This time also fix the capstone.h header check to use the path we're
already patching in.

Bug: https://bugs.gentoo.org/873157
diff --git a/include/disas/capstone.h b/include/disas/capstone.h
index e29068dd97..d8fdc5d537 100644
--- a/include/disas/capstone.h
+++ b/include/disas/capstone.h
@@ -3,7 +3,7 @@
 
 #ifdef CONFIG_CAPSTONE
 
-#include <capstone.h>
+#include <capstone/capstone.h>
 
 #else
 
diff --git a/meson.build b/meson.build
index 20fddbd707..50ce96bbd1 100644
--- a/meson.build
+++ b/meson.build
@@ -2605,7 +2605,7 @@ if not get_option('capstone').auto() or have_system or have_user
   # that reports a wrong -I path, causing the #include to
   # fail later. If the system has such a broken version
   # do not use it.
-  if capstone.found() and not cc.compiles('#include <capstone.h>',
+  if capstone.found() and not cc.compiles('#include <capstone/capstone.h>',
                                           dependencies: [capstone])
     capstone = not_found
     if get_option('capstone').enabled()
